kasa/smartdevice.py

```diff
@@ -454,14 +467,14 @@ def hw_info(self) -> Dict:
             "oemId",
             "dev_name",
         ]
-        sys_info = self.sys_info
+        sys_info = self._sys_info
         return {key: sys_info[key] for key in keys if key in sys_info}
 
     @property  # type: ignore
     @requires_update
     def location(self) -> Dict:
         """Return geographical location."""
-        sys_info = self.sys_info
+        sys_info = self._sys_info
```

Comment:
We need a better solution than rewriting all these accesses. Alas, I don't currently have time to investigate so ideas are welcome.

Reply:
I think this is a good solution to allowing `update_from_discover_info()` to work from the 20002 devices.  It still allows consumers to access certain fields without raising a requires_update exception.

Reply:
But but, couldn't we just initialize the `_sys_info` in `_set_sys_info` to avoid renaming things?

Reply:
Ok, that's fine, you got me convinced! :+1: